airflow/providers/microsoft/azure/hooks/container_instance.py

```diff
@@ -19,7 +19,7 @@
 
 import warnings
 from functools import cached_property
-from typing import TYPE_CHECKING, Any
+from typing import TYPE_CHECKING, Any, cast
 
 from azure.common.client_factory import get_client_from_auth_file, get_client_from_json_dict
 from azure.common.credentials import ServicePrincipalCredentials
```

Comment:
Package is deprecated and no longer working. Consider using `azure.identity` ([relevant info](https://learn.microsoft.com/en-us/python/api/azure-common/?view=azure-python-previous))

Related issue on [azure-sdk-for-python](https://github.com/Azure/azure-sdk-for-python/issues/16908#issuecomment-785216591)

Reply:
Removed ServicePrincipalCredentials and added ClientSecretCredential from `azure.identity`